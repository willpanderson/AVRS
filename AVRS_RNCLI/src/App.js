import * as React from 'react';
import { View, Text, TouchableOpacity, Image, Button, StyleSheet } from 'react-native';

import HomePage from './HomePage';
import MainMenu from './MainMenu';
import { func } from 'prop-types';


class App extends React.Component {
    state = {
        isLoggedIn: true,
    };

    logInOut = () => {
        let loggedIn = !this.state.isLoggedIn;
        this.setState({
            isLoggedIn: loggedIn,
        });
        console.log(this.state.isLoggedIn);
    };

    render() {
        if (this.state.isLoggedIn == true) {
            return <MainMenu logInChange={this.logInOut}/>
        } else {
            return (
                <HomePage logInChange={this.logInOut} />
            )
        }

    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#377888',
    },
    image: {
        flex: 2,
        width: 250,
        resizeMode: 'contain',
    },
    buttons: {
        flex: 1,
    },
    button: {
        width: 200,
        height: 50,
        backgroundColor: '#05111B',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 10,
        margin: 15,
    },
    buttonText: {
        color: '#DBDDDF',
    },
})

export default App;