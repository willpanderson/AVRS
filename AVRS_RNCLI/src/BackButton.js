import React from 'react';
import PropTypes from 'prop-types';

import { Pressable, Text, View, StyleSheet } from 'react-native';

class BackButton extends React.Component {

    static propTypes = {
        backPress: PropTypes.func.isRequired,
    }

    render() {
        return (
            <View>
                <Pressable 
                    style={({ pressed }) => [
                        styles.button,
                        pressed ? { opacity: 0.8 } : {},
                    ]}
                    onPress={this.props.backPress}>
                    <Text>Back</Text>
                </Pressable>
            </View>
    );
  }
}

const styles = StyleSheet.create({
    button: {
        width: 50,
        backgroundColor: '#377888',
    },

})

export default BackButton;
