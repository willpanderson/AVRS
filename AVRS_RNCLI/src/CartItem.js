import {View, Text, StyleSheet, Pressable} from 'react-native';
import React, {useState} from 'react';

export default function CartItem(props) {
  const [count, setCount] = useState(props.item.count);

  const onMinusPress = () => {
    console.log('clicked minus');
    setCount(count - 1);
    props.item.count = count - 1;
    console.log(props.item.count);
  };

  const onPlusPress = () => {
    console.log('clicked plus');
    setCount(count + 1);
    props.item.count = count + 1;
  };

  if (props.item.count > 0) {
    return (
      <View style={styles.container}>
        <View style={styles.buttonsContainer}>
          <Pressable
            style={({pressed}) => [
              styles.button,
              pressed ? {opacity: 0.8} : {},
            ]}
            onPress={() => onPlusPress()}>
            <Text style={styles.buttonText}>+</Text>
          </Pressable>

          <Text>{count}</Text>

          <Pressable
            style={({pressed}) => [
              styles.button,
              pressed ? {opacity: 0.8} : {},
            ]}
            onPress={() => onMinusPress()}>
            <Text style={styles.buttonText}>-</Text>
          </Pressable>
        </View>

        <View style={styles.nameContainer}>
          <Text>{props.item.name}</Text>
        </View>
      </View>
    );
  }
  return <></>;
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    // alignItems: 'flex-start',
    margin: 20,
  },
  buttonsContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'row',
  },
  nameContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  button: {
    backgroundColor: '#05111B',
    width: 30,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 15,
    transform: [{scaleX: 1.5}],
    marginHorizontal: 15,
  },
  buttonText: {
    color: '#DBDDDF',
  },
});
