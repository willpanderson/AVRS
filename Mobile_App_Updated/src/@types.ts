export type RootStackParamList = {
  splash: undefined;
  auth: undefined;
  tabMenu: undefined;
};

export type AuthContextType = {
  signIn: (email: string, password: string) => void;
  signOut: () => void;
  signUp: (email: string, password: string) => void;
};

export type AuthContextTypeTwo = {
  signIn: (
    email: string,
    password: string,
    dispatch: (arg0: {type: string; token: string}) => void,
    storeUserSession: () => void,
  ) => void;
  signOut: (
    removeItem: (arg0: string) => void,
    dispatch: (arg0: {type: string; token: undefined}) => void,
  ) => void;
  signUp: (
    email: string,
    password: string,
    dispatch: (arg0: {type: string; token: string}) => void,
    storeUserSession: () => void,
  ) => void;
};

export type CartItemType = {
  name: string;
  count: number;
};

export type CartContextType = {
  addToCart: (item: CartItemType) => void;
  addCartItem: (index: number) => void;
  minusCartItem: (index: number) => void;
  deleteFromCart: (item: CartItemType) => void;
  emptyCart: () => void;
  cart: Array<CartItemType>;
};

export type SnackType = {
  Name: string;
  Reserved: number;
  Total: number;
  image: string;
};

export interface ParentCompProps {
  children?: React.ReactNode;
}
