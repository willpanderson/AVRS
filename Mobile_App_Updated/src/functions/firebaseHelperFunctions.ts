import {auth} from '../../firebase';
import {
  signInWithEmailAndPassword,
  createUserWithEmailAndPassword,
  signOut,
  User,
  deleteUser,
} from 'firebase/auth';

export const createAccount = async (email: string, password: string) => {
  return createUserWithEmailAndPassword(auth, email, password)
    .then(userCredential => {
      // Signed in
      const user = userCredential.user;
      console.log(user);
      return true;
      // ...
    })
    .catch(error => {
      const errorCode = error.code;
      const errorMessage = error.message;
      console.log('Error code: ' + errorCode);
      console.log('Error message: ' + errorMessage);
      return false;
      // ..
    });
};

export const logIn = async (email: string, password: string) => {
  return signInWithEmailAndPassword(auth, email, password)
    .then(userCredential => {
      const user = userCredential.user;
      console.log('user');
      console.log(user);
      return true;
    })
    .catch(error => {
      const errorCode = error.code;
      const errorMessage = error.message;
      console.log('Error code: ' + errorCode);
      console.log('Error message: ' + errorMessage);
      return false;
    });
};

export const handleSignOut = async () => {
  return signOut(auth)
    .then(() => {
      console.log('Sign out');
      return true;
    })
    .catch(error => {
      console.log(error.message);
      return false;
    });
  // return true;
};

export const deleteAccount = async () => {
  const user = auth.currentUser as User;
  console.log(user);
  return deleteUser(user)
    .then(() => {
      return true;
    })
    .catch(error => {
      console.log(error.message);
      return false;
    });
};
