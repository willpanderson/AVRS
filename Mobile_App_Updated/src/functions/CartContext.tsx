import React from 'react';
import {CartContextType, CartItemType, ParentCompProps} from '../@types';

const CartContext = React.createContext<CartContextType | null>(null);

export const useCart = () => React.useContext(CartContext);

const CartProvider: React.FC<ParentCompProps> = ({children}) => {
  const [cart, setCart] = React.useState<CartItemType[]>([]);

  const addToCart = (item: CartItemType) => {
    console.log('add to cart ' + item);
    setCart([...cart, item]);
    console.log(cart);
  };

  const addCartItem = (index: number) => {
    console.log('add 1 to count for ' + cart[index].name);
    cart[index].count++;
    console.log(cart[index].count);
  };

  const minusCartItem = (index: number) => {
    console.log('subtract 1 to count for ' + cart[index].name);
    cart[index].count--;
    console.log(cart[index].count);
    if (cart[index].count < 1) {
      deleteFromCart(cart[index]);
    }
  };

  const deleteFromCart = (item: CartItemType) => {
    console.log('Deleting ' + item.name + ' from cart');
    setCart(cart.filter(items => items.name !== item.name));
    console.log(cart);
  };

  const emptyCart = () => {
    setCart([]);
    console.log('Empty cart');
    console.log(cart);
  };

  return (
    <CartContext.Provider
      value={{
        cart,
        addToCart,
        addCartItem,
        minusCartItem,
        deleteFromCart,
        emptyCart,
      }}>
      {children}
    </CartContext.Provider>
  );
};

export default CartProvider;
