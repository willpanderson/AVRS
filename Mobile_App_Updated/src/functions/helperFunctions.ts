import {CartItemType} from '../@types';

export const onButtonPress = (
  outsideOpperation: (arg0: any) => void,
  insideOpperation: (arg0: any) => void,
  item: number,
  list: Array<CartItemType>,
) => {
  console.log('Clicked button');
  outsideOpperation(item);
  insideOpperation(list[item].count);
};
