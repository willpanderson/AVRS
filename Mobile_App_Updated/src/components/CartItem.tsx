import {View, Text, StyleSheet, Pressable} from 'react-native';
import React, {useState} from 'react';
import {CartContextType} from '../@types';
import {useCart} from '../functions/CartContext';
import {onButtonPress} from '../functions/helperFunctions';

export default function CartItem(props: {item: number; key: number}) {
  const {cart, addCartItem, minusCartItem} = useCart() as CartContextType;

  const [count, setCount] = useState(cart[props.item].count);

  if (cart[props.item].count > 0) {
    return (
      <View style={styles.container}>
        <View style={styles.buttonsContainer}>
          <Pressable
            style={({pressed}) => [
              styles.button,
              pressed ? {opacity: 0.8} : {},
            ]}
            onPress={() =>
              onButtonPress(addCartItem, setCount, props.item, cart)
            }>
            <Text style={styles.buttonText}>+</Text>
          </Pressable>
          <Text>{count}</Text>

          <Pressable
            style={({pressed}) => [
              styles.button,
              pressed ? {opacity: 0.8} : {},
            ]}
            onPress={() =>
              onButtonPress(minusCartItem, setCount, props.item, cart)
            }>
            <Text style={styles.buttonText}>-</Text>
          </Pressable>
        </View>

        <View style={styles.nameContainer}>
          <Text>{cart[props.item].name}</Text>
        </View>
      </View>
    );
  }
  return <></>;
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    // alignItems: 'flex-start',
    margin: 20,
  },
  buttonsContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'row',
  },
  nameContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  button: {
    backgroundColor: '#05111B',
    width: 30,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 15,
    transform: [{scaleX: 1.5}],
    marginHorizontal: 15,
  },
  buttonText: {
    color: '#DBDDDF',
  },
});
