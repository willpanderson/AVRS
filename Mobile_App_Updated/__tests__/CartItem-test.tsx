import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CartItem from '../src/components/CartItem';
import * as CartProviderr from '../src/functions/CartContext';
import {CartItemType} from '../src/@types';
import {fireEvent, render} from '@testing-library/react-native';

describe('Tests for CartItem component.', () => {
  // Mocked CartContext
  const cart = [{name: 'snack', count: 2}];
  const addCartItem = jest.fn((index: number) => {
    console.log(index);
    cart[index].count++;
  });
  const minusCartItem = jest.fn((index: number) => {
    console.log(index);
    cart[index].count--;
  });
  const addToCart = (item: CartItemType) => console.log(item);
  const deleteFromCart = (item: CartItemType) => console.log(item);
  const emptyCart = () => console.log('empty');

  // THIS ONE WORKED!!!
  // This is what actually mocks the useCart hook
  jest.spyOn(CartProviderr, 'useCart').mockImplementation(() => ({
    cart,
    addCartItem,
    minusCartItem,
    addToCart,
    deleteFromCart,
    emptyCart,
  }));

  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <CartProviderr.default>
          <CartItem item={0} key={0} />
        </CartProviderr.default>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Add a snack', () => {
    const {getByText} = render(
      <CartProviderr.default>
        <CartItem item={0} key={0} />
      </CartProviderr.default>,
    );
    let initial = cart[0].count;
    fireEvent.press(getByText('+'));
    let later = cart[0].count;
    expect(addCartItem).toBeCalled();
    expect(later > initial).toBe(true);
  });

  it('Subtract a snack', () => {
    const {getByText} = render(
      <CartProviderr.default>
        <CartItem item={0} key={0} />
      </CartProviderr.default>,
    );

    let initial = cart[0].count;
    let countDisplayed = getByText(initial.toString());
    expect(countDisplayed).toBeTruthy();

    fireEvent.press(getByText('-'));

    let later = cart[0].count;
    countDisplayed = getByText(later.toString());
    expect(countDisplayed).toBeTruthy();

    expect(minusCartItem).toBeCalled();
    expect(later < initial).toBe(true);
  });
});
