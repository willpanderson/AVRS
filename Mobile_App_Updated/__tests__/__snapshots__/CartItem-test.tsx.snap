// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CartItem component. Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#05111B",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 15,
            "transform": Array [
              Object {
                "scaleX": 1.5,
              },
            ],
            "width": 30,
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#DBDDDF",
          }
        }
      >
        +
      </Text>
    </View>
    <Text>
      2
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#05111B",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "marginHorizontal": 15,
            "transform": Array [
              Object {
                "scaleX": 1.5,
              },
            ],
            "width": 30,
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#DBDDDF",
          }
        }
      >
        -
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      snack
    </Text>
  </View>
</View>
`;
